spanner/spansql/parser.go

```diff
@@ -1286,6 +1289,50 @@ func (p *parser) parseAlterTable() (*AlterTable, *parseError) {
 	}
 }
 
+func (p *parser) parseAlterDatabase() (*AlterDatabase, *parseError) {
+	debugf("parseAlterDatabase: %v", p)
+
+	/*
+			ALTER DATABASE database_id
+		    	action
+
+			action:
+		    SET OPTIONS ( optimizer_version = { 1 ...  2 | null },
+		                  version_retention_period = { 'duration' | null },
+		                  enable_key_visualizer = { true | false | null } )
+	*/
+
+	if err := p.expect("ALTER"); err != nil {
+		return nil, err
+	}
+	pos := p.Pos()
+	if err := p.expect("DATABASE"); err != nil {
+		return nil, err
+	}
+	// The following method is also valid for database names.
+	dbname, err := p.parseTableOrIndexOrColumnName()
+	if err != nil {
+		return nil, err
+	}
+	a := &AlterDatabase{Name: dbname, Position: pos}
+
+	tok := p.next()
+	if tok.err != nil {
+		return nil, tok.err
+	}
+	switch {
+	default:
```

Comment:
nits: can we move default to the last one in `switch`?

Reply:
I would prefer to keep it as the first element in the switch for the sake of consistency. This is how all parsing is implemented in spansql. See for example [ALTER TABLE](https://github.com/googleapis/google-cloud-go/blob/2cb9ad72416c84742cca4073e7583d4e0b02cb4f/spanner/spansql/parser.go#L1217).

This structure also makes it clear at the start of the switch block what can be expected in the rest of the block.